### DISCUSSION TITLE
How do we coordinate?

### DISCUSSION THREAD

**Discussion by mitko** [mitko/readable_climate_reports#6] [General]
to: @petermr @rowanc1 @yipperbore

Hello friends. I'm going to bring up the major coordination questions we may face, so we can set the base for this project.

1. Mode of collaboration.
  * We are in various time zones and have jobs, so 
  * I would suggest **most collaboration to happen asynchronously, via Issues/PRs/pushing code/Discussions**
  * If you are open to it - we can also set up a quick video chat to get to know each other
  
2. Technical decisions
   * We are looking at several different technologies at the moment. Not settled on which we want to use.
   * I suggest we start a separate discussion thread about overall architecture. 
   * We don't have to make a decision yet. We can work independently initially, and merge/push freely to the repo. 
   * This way we can test out various technologies, depending on what each of us is excited about.
   * If a certain approach looks promising, and we want to make it "official", let's start a new discussion about it, perhaps with some demo or quality assessment. 
   * If we run into trouble, we can again start a new discussion.
   
 3. Adding new people to the project
    * this project will be open to other collaborators.
    * they will need to "buy in" to whatever mode of collaboration and architecture we choose. 
   
Please comment on

**Comment by petermr** [comment]
Thanks!



On Wed, Apr 13, 2022 at 3:44 AM Dimitar Simeonov ***@***.***>
wrote:

>
>    1. Mode of collaboration.
>
>
>    - We are in various time zones and have jobs, so
>    - I would suggest *most collaboration to happen asynchronously, via
>    Issues/PRs/pushing code/Discussions*
>
> Agreed

>
>    - If you are open to it - we can also set up a quick video chat to get
>    to know each other
>
> Agreed. Useful to set out some of the issues we may need to discuss. This
is a complex project and  there will be several ways of tackling it. We
should avoid being didactic

>
>
>
>    1.
>
>    Technical decisions
>    - We are looking at several different technologies at the moment. Not
>       settled on which we want to use.
>
> I think there will often be multiple software tools, but the data
structures should be communal if possible


>
>    1.
>       - I suggest we start a separate discussion thread about overall
>       architecture.
>
> This is the most important technical decision. The architecture should
drive the software, not the other way round. And we should always think
about abstraction (e.g. XML vs JSON is less important than the abstract
data structure)

>
>    1.
>       - We don't have to make a decision yet. We can work independently
>       initially, and merge/push freely to the repo.
>
> Agreed. We don't know what is going to work until we try it.

>
>    1.
>       - Adding new people to the project ; this project will be open to
>       other collaborators; they will need to "buy in" to whatever mode of
>       collaboration and architecture we choose.
>
>
I think we need to agree our communal goals or we will get broad and pull
in different directions

—

> Reply to this email directly, view it on GitHub
> <https://github.com/mitko/readable_climate_reports/discussions/6>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAFTCSZ4XP357RFRIQCCOI3VEYYG3ANCNFSM5TJHETDQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
Peter Murray-Rust
Founder ContentMine.org
and
Reader Emeritus in Molecular Informatics
Dept. Of Chemistry, University of Cambridge, CB2 1EW, UK

**Comment by yipperbore** [comment]
I agree working towards a live chat on Zoom would likely help organize a bit, we can discuss what is in each other’s wheel house so we can divide and conquer!

Midwest Glass Fabricators
Office (248) 889-7900
Mobile (248) 249-0478
***@***.******@***.***> | www.mwgf.com<http://www.mwgf.com/>
